test/e2e/gc_e2e_test.go

```diff
@@ -34,159 +33,257 @@ var _ = Describe("Garbage collector", func() {
-		require.NoError(GinkgoT(), err)
-
-		// wait for deletion of ownerB
-		require.NoError(GinkgoT(), waitForDelete(func() error {
-			_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), ownerB.GetName(), metav1.GetOptions{})
-			return err
-		}))
-
-		// check for dependent (should be deleted since last blocking owner was deleted)
-		_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(context.TODO(), dependent.GetName(), metav1.GetOptions{})
-		require.Error(GinkgoT(), err)
-		require.True(GinkgoT(), k8serrors.IsNotFound(err))
-		GinkgoT().Log("dependent successfully garbage collected after both owners were deleted")
+	// TestOwnerReferenceGCBehavior runs a simple check on OwnerReference behavior to ensure
+	// a resource with multiple OwnerReferences will not be garbage collected when one of its
+	// owners has been deleted.
+	// Test Case:
+	//				CSV-A     CSV-B                        CSV-B
+	//				   \      /      --Delete CSV-A-->       |
+	//				   ConfigMap						 ConfigMap
+	Context("Given a dependent resource associated with multiple owners", func() {
+		var (
+			ownerA   v1alpha1.ClusterServiceVersion
+			ownerB   v1alpha1.ClusterServiceVersion
+			fetchedA *v1alpha1.ClusterServiceVersion
+			fetchedB *v1alpha1.ClusterServiceVersion
+
+			dependent *corev1.ConfigMap
+
+			propagation metav1.DeletionPropagation
+			options     metav1.DeleteOptions
+		)
+
+		BeforeEach(func() {
+
+			ownerA = newCSV("ownera", testNamespace, "", semver.MustParse("0.0.0"), nil, nil, newNginxInstallStrategy("dep-", nil, nil))
+			ownerB = newCSV("ownerb", testNamespace, "", semver.MustParse("0.0.0"), nil, nil, newNginxInstallStrategy("dep-", nil, nil))
+
+			// create all owners
+			var err error
+			fetchedA, err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Create(context.TODO(), &ownerA, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			fetchedB, err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Create(context.TODO(), &ownerB, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			dependent = &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dependent",
+				},
+				Data: map[string]string{},
+			}
+
+			// add owners
+			ownerutil.AddOwner(dependent, fetchedA, true, false)
+			ownerutil.AddOwner(dependent, fetchedB, true, false)
+
+			// create ConfigMap dependent
+			_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Create(context.TODO(), dependent, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred(), "dependent could not be created")
+
+			propagation = metav1.DeletePropagationForeground
+			options = metav1.DeleteOptions{PropagationPolicy: &propagation}
+		})
+
+		When("removing one of the owner using 'Foreground' deletion policy", func() {
```

Comment:
Might be useful to include a link that describes the `Foreground` deletion policy.

Reply:
Is it okay to include links to external docs in source code?